test/spec/openapi/refs.js

```diff
@@ -282,3 +282,39 @@ test('uses examples if has property required in body', async (t) => {
   t.ok(schema.parameters)
   t.same(schema.parameters[0].in, 'query')
 })
+
+test('support absolute $ref inside the json-schema definitions', async (t) => {
+  const fastify = Fastify()
+
+  await fastify.register(fastifySwagger, openapiOption)
+  fastify.register(async (instance) => {
+    instance.addSchema({
+      $id: 'NestedSchema',
+      definitions: {
+        SchemaA: {
+          type: 'object',
+          properties: {
+            id: { type: 'string' }
+          }
+        },
+        SchemaB: {
+          type: 'object',
+          properties: {
+            example: { $ref: 'NestedSchema#/definitions/SchemaA' }
```

Comment:
@asc11cat Will it work with a local $ref `#/definitions/SchemaA'`?

Reply:
> @asc11cat Will it work with a local $ref `#/definitions/SchemaA'`?

@ivan-tymoshenko, won't work, it seems like we lack support for the relative path overall as I mentioned in https://github.com/fastify/fastify-swagger/issues/675#issuecomment-1272280205

Should we add some more magic with replaces, or can there be a better way?

Reply:
I wrote json schema RefResolver for FJS. I don't know if it work here, because swagger/openapi has some specific rules. You can check if it work here.
https://github.com/fastify/fast-json-stringify/blob/master/lib/ref-resolver.js

Reply:
It has two public methods `addSchema` and `getSchema`. If you have any questions, feel free to ask.